File: 2000 98 & ME/QuickStart/howto/samples/Reflection/gettypes/vb/gettypes.vb
'
'  This program get's all the types from a system assembly and this assembly.
'


Imports System
Imports System.Reflection

Public Class GetTypes

	
    Public Shared Sub Main()
        Dim o As Object
        'List all the types from the mscorlib assembly.
		
        'Get the mscorlib assembly, it's the one Object is defined in
        Dim a As reflection.Assembly = GetType(Object).Module.Assembly
		
		'Get all the types in this assembly
		Dim types() As Type = a.GetTypes()
		
		'let's take a look at them, and gather a little bit of data as we do it.
		Dim numValueTypes As Integer = 0
		Dim numInterfaces As Integer = 0
		Console.WriteLine ("Get all the types from the assembly: '0End'", a.GetName())
        For Each o In types
            Dim t As Type = CType(o, Type)
			Console.WriteLine(t.FullName)
            If (t.IsInterface) Then
                numInterfaces += 1
            End If
            If (t.IsValueType) Then
                numValueTypes += 1
            End If
		Next
		Console.WriteLine ("Out of {0} End types, {1} End are interfaces and {2} End are value types", _
			           types.Length, _
                       numInterfaces, _
                       numValueTypes)
		Console.WriteLine 
		
		'Load an assembly from disk, just so happens it's THIS assembly
        Dim b As Reflection.Assembly = Reflection.Assembly.LoadFrom("gettypes.exe")
		Dim types2() As Type = b.GetTypes()
        For Each o In types2
            Dim t As Type = CType(o, Type)
			Console.WriteLine (t.FullName)
		Next	
		
        Console.WriteLine()
        Console.WriteLine("Press Return to exit.")
		Console.Read()
	End Sub
	
End Class

Public class TestClass1

End Class

Public Class TestClass2

End Class

File: 2000 98 & ME/QuickStart/howto/samples/Reflection/gettypes/gettypes.cs
//
//  This program get's all the types from a system assembly and this assembly.
//


using System;
using System.Reflection;

class GetTypes
{
	
	public static void Main(String[] args)
	{
		//List all the types from the mscorlib assembly.
		
		//Get the mscorlib assembly, it's the one Object is defined in
		Assembly a = typeof(Object).Module.Assembly;
		
		//Get all the types in this assembly
		Type [] types = a.GetTypes ();
		
		//let's take a look at them, and gather a little bit of data as we do it.
		int numValueTypes = 0;
		int numInterfaces = 0;
		Console.WriteLine ("Get all the types from the assembly: '{0}'", a.GetName());
		foreach (Type t in types)
		{
			Console.WriteLine (t.FullName);
			if (t.IsInterface) numInterfaces++;
			if (t.IsValueType) numValueTypes++;
		}
		Console.WriteLine ("Out of {0} types, {1} are interfaces and {2} are value types",
			types.Length, numInterfaces, numValueTypes);
		
		//Load THIS assembly using its file name
		Assembly b = Assembly.LoadFrom ("GetTypes.exe");
		Type [] types2 = b.GetTypes ();
		foreach (Type t in types2)
		{
			Console.WriteLine (t.FullName);
		}
		
		Console.WriteLine ("\r\nPress Return to exit.");
		Console.Read();
	}
	
}

public class TestClass1
{
}

public class TestClass2
{
}